File: frontend/types/corridor.ts
export interface Hospital {
  id: string;
  name: string;
  lat: number;
  lng: number;
}

export interface RouteVariant {
  geometry: {
    type: "LineString";
    coordinates: [number, number][]; // Array of [lng, lat]
  };
  duration_minutes: number;
  label: string;
  note?: string;
  resource_note?: string;
}

export interface CorridorRoutes {
  fastest: RouteVariant;
  safest: RouteVariant;
  protected: RouteVariant;
}

export interface CorridorResponse {
  hospital_id: string;
  hospital_name: string;
  incident_junction_id: string;
  incident_junction_name: string;
  is_approximate: boolean;
  routes: CorridorRoutes;
}

export interface CorridorPlanRequest {
  hospital_id: string;
  incident_junction_id: string;
}

File: backend/app/schemas/corridor.py
from __future__ import annotations

from typing import Any, Optional
from pydantic import BaseModel


class HospitalResponse(BaseModel):
    id: str
    name: str
    lat: float
    lng: float


class CorridorRequest(BaseModel):
    hospital_id: str
    incident_junction_id: str


class RouteVariant(BaseModel):
    geometry: dict[str, Any]  # GeoJSON LineString geometry
    duration_minutes: int
    label: str
    note: Optional[str] = None
    resource_note: Optional[str] = None


class CorridorRoutes(BaseModel):
    fastest: RouteVariant
    safest: RouteVariant
    protected: RouteVariant


class CorridorResponse(BaseModel):
    hospital_id: str
    hospital_name: str
    incident_junction_id: str
    incident_junction_name: str
    is_approximate: bool
    routes: CorridorRoutes
